dvc/commands/experiments/__init__.py

```diff
@@ -58,3 +58,43 @@ def add_parser(subparsers, parent_parser):
     for cmd in SUB_COMMANDS:
         cmd.add_parser(experiments_subparsers, parent_parser)
     fix_plumbing_subparsers(experiments_subparsers)
+
+
+def add_rev_selection_flags(
+    experiments_subcmd_parser, command: str, default: bool = True
+):
+    experiments_subcmd_parser.add_argument(
+        "-A",
+        "--all-commits",
+        action="store_true",
+        default=False,
+        help=(
+            f"{command} all experiments in the repository "
+            "(overrides `--rev` and `--num`)."
+        ),
+    )
+    default_msg = " (HEAD by default)" if default else ""
+    msg = (
+        f"{command} experiments derived from the specified `<commit>` as "
+        f"baseline{default_msg}."
+    )
+    experiments_subcmd_parser.add_argument(
+        "--rev",
+        type=str,
+        default=None,
+        help=msg,
+        metavar="<commit>",
+    )
+    experiments_subcmd_parser.add_argument(
+        "-n",
+        "--num",
+        type=int,
+        default=1,
+        dest="num",
+        metavar="<num>",
+        help=(
+            f"{command} experiments from the `--rev` baseline and from `num` "
+            "commits before it. Give a negative value to include all the "
+            "parent commits."
```

Comment:
This probably needs an update similar to https://github.com/iterative/dvc.org/pull/3187#pullrequestreview-883999298 ?

Reply:
Also, I have a question: what happens when `--rev` is a merge? Which ancestors are included by the `<num>`?

Reply:
It will include the first parent for merge commits (linear ancestors, equivalent to using tilde `~` in git). We don't support specifying a different parent (i.e carat `^` in git).

So for `--rev HEAD -n 2` You would get the commit referenced by `HEAD~2` in git (as opposed to `HEAD~2^2/HEAD~2^3...`

see: https://stackoverflow.com/a/43046393/1538451

Reply:
Got it, thanks @pmrowla ! I wonder if we want to include this info in the description or in docs. May be relevant if people use large `-n` values (likely to include merges).